File: 02/chess.js
for (var i = 1; i <= 8; i++) {
  var string = '';
  var sharp = "#";
  var blank = " ";
  for (var j = 1; j <= 8; j++) {
    if (j % 2 != 0) {
      string += (i % 2 != 0) ? sharp : blank;
    } else {
      string += (i % 2 == 0) ? sharp : blank;
    }
  }
  console.log(string);
}

File: 02/chess.go
package main

import (
  "fmt"
  "bytes"
)

func main() {
  str := ""
  sharp := "#" 
  black := " "
  // naive concate
  for i := 1; i <= 8; i++ {
    str = "" 
    for j := 1; j <= 8; j++ {
      if j % 2 == 0 {
        str += sharp
      } else {
        str += black
      }
    }
    sharp, black = black, sharp
    fmt.Println(str)
  }
  // append concate
  for i := 1; i <= 8; i++ {
    s := []string{}
    for j := 1; j <= 8; j++ {
      if j % 2 == 0 {
        s = append(s, sharp)
      } else {
        s = append(s, black)
      }
    }
    sharp, black = black, sharp
    fmt.Println(s)
  }
  // buffer
  for i := 1; i <= 8; i++ {
    var buffer bytes.Buffer
    for j := 1; j <= 8; j++ {
      if j % 2 == 0 {
        buffer.WriteString(sharp)
      } else {
        buffer.WriteString(black)
      }
    }
    sharp, black = black, sharp
    fmt.Println(buffer.String())
  }
}